### PR TITLE
燃果-谭静-前端

### PR DESCRIPTION
**Pull request by tanjing9406**
1. 新增组件 Carousel；
2. 新增单元测试
3. 新增 readme
3. 构建项目，[在线 Demo](https://tanjing9406.github.io/interview-assignments/);
4. [我的简历](https://tanjing9406.github.io/interview-assignments/my-resume-tanjing.pdf)

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+        </article>
+      </main>
+      <main className="main-wrap w-full h-full airpods">
+        <header className="title">
+          Buy a Tablet or xPhone for college.<br />
+          Get airPods.
+        </header>
+      </main>
+    </Carousel>
+  </div>;
```
Comment (YuBoxiang) [inline frontend/src/App.tsx:32]: 可否数据化一下？

File: frontend/src/components/carousel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import "./index.css"
+import { getIdentifierAnimationStyle, getSlickPositionLeft } from "./tools";
+
+export interface Props {
+    children: Array<ReactElement>, // the children's length should be more than 1
+    duration?: number // unit is millisecond, default value is 2500
+    className?: string
+}
+
+let timer: any
```
Comment (YuBoxiang) [inline frontend/src/components/carousel/index.tsx:11]: 可否使用更准确的 类型替代any？

File: frontend/src/App.tsx (hunk 1 above)
Comment (tanjing9406) [reply frontend/src/App.tsx]: done. adding Slider component for it. thx.

File: frontend/src/components/carousel/index.tsx (hunk 2 above)
Comment (tanjing9406) [reply frontend/src/components/carousel/index.tsx]: done. number Return Type for it. thx very much for your comment!

**Pull request closed without merging**